quickwit/quickwit-metastore/src/metastore/mod.rs

```diff
@@ -229,14 +241,137 @@ pub trait Metastore: Send + Sync + 'static {
         index_id: &str,
         opstamp_start: u64,
     ) -> MetastoreResult<Vec<DeleteTask>>;
+}
 
-    /// Lists splits with `split.delete_opstamp` < `delete_opstamp` for a given `index_id`.
-    /// These splits are called "stale" as they have an `delete_opstamp` strictly inferior
-    /// to the given `delete_opstamp`.
-    async fn list_stale_splits(
-        &self,
-        index_id: &str,
-        delete_opstamp: u64,
-        num_splits: usize,
-    ) -> MetastoreResult<Vec<Split>>;
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+/// A filter builder for filtering splits within the metastore.
+pub struct SplitFilter<'a> {
```

Comment:
I'm tempted to go for the former as it feels more resilient to change if we ever decide that we want to be able to list all splits from all indexes. But I'm not sure if I would go with the name `ListSplitRequest` as it feels more GRPC related rather than a db query essentially, maybe `ListSplitQuery`?

Reply:
Yes, `ListSplitQuery` is great.